cpc/criterion/soft_align.py

```diff
@@ -323,4 +323,16 @@ def forward(self, cFeature, encodedData, label):
         outLossesD = outLosses.detach()
         losses = losses.mean() / outLossesD.sum() * outLossesD
 
-        return losses, outAcc
+        captureRes = None
+        if captureOptions != None:
+            for o in captureOptions:
+                assert o in ('pred', 'align')
+            captureRes = {}
+            if 'pred' in captureOptions:
+                # need to remove 'predict self loop' stuff from captured data
+                captureRes['pred'] = predictions[:,:,:,1:]
```

Comment:
I am unsure about this [1:] at the end - is it to skip the self-loop?

Reply:
Yeah, I was removing it as it's not really a prediction, but forgot it would spoil alignment indices - fixed, now self loop is also saved in predictions